CONTRIBUTING.md

```diff
@@ -0,0 +1,163 @@
+under the same terms as the project's overall 'outbound' license - in this case,
+Apache Software License v2 (see [LICENSE](LICENSE)).
+
+To simplify project management, Matrix iOS SDK and Matrix iOS Kit issues are
+managed in the Element-iOS
+[repository](https://github.com/vector-im/element-ios/issues).
+
+## How to contribute
+
+The preferred and easiest way to contribute changes to the project is to fork
+it on GitHub, and then [create a pull request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) to ask us to pull your changes
+into our repo.
+
+We use GitHub's pull request workflow to review the contribution, and either
+ask you to make any refinements needed or merge it and make them ourselves.
+
+Things that should go into your PR description:
+
+- References to any bugs fixed by the change (in GitHub's `Fixes` [notation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
+- Notes for the reviewer that might help them to understand why the change
+is necessary or how they might better review it
+- Screenshots or videos if applicable
+
+Your PR must also:
+
+- be based on the develop branch
+- include a changelog file entry (see [below](#changelog))
+- include a [sign off](#sign-off)
+
+## Attribution
+
+Everyone who contributes anything to Matrix is welcome to be listed in the
+[AUTHORS.rst](AUTHORS.rst) file for the project in question. Please feel free
+to include a change to AUTHORS.rst in your pull request to list yourself and
+a short description of the area(s) you've worked on. Also, we sometimes have
+swag to give away to contributors - if you feel that Matrix-branded apparel is
+missing from your life, please mail us your shipping address to <em>matrix at
+matrix.org</em> and we'll try to fix it :).
+
+## Changelog
+
+All changes, even minor ones, need a corresponding changelog / newsfragment
+entry. These are managed by [Towncrier](https://github.com/twisted/towncrier).
+
+To create a changelog entry, make a new file in the `changelog.d` directory
+named in the format of `ElementIOSIssueNumber.type`. The type can be one of the
+following:
+
+- `feature` for a new feature
+- `change` for updates to an existing feature
+- `bugfix` for bug fix
+- `api` for an api break
+- `i18n` for translations
+- `build` for changes related to build, tools, CI/CD
+- `doc` for updates to the documentation
+- `misc` for other changes
+
+This file will become part of our [changelog](CHANGES.md) at the next
+release, so the content of the file should be a short description of your
+change in the same style as the rest of the changelog. The file can contain
+Markdown formatting. It should start with the area of the change (screen,
+module, ...) and end with a full stop (.) or an exclamation mark (!) for
+consistency.
```

Comment:
Is there a limitation on how many lines the description can be in the file? Possibly worth mentioning that in here too if it needs to stay on a single line.